### DISCUSSION TITLE
Monster corpses disappearing quickly

### DISCUSSION THREAD

**Discussion by Goldeye** [FakeFishGames/Barotrauma#13937] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

Monster corpses disappear too quickly (i.e. after a minute or two) inside and outside the sub, often before I can reasonably get to them to loot.  Some corpses do persist longer.

### Reproduction steps

Sorry, it has happened this entire single player campaign (started 1 or 2 hotfixes ago) and I have no idea what's causing it.  Let me know anything I should check.

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Single player

### Version

v1.4.6.0 (Blood in the Water Update, hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by pavel4win** [comment]
I am completely agree!
Why do monster corpses disappear anyway?
It's just not fair.
This is unpleasant and deprives the game of realism.
Several times it happened that monsters disappeared right before my eyes.
Sometimes you are just about to go out for loot, and the monster has already disappeared.
It is necessary to make sure that only the monsters from which you collected loot disappear. 
Or who are 500 meters away from the boat.
And it’s best to make a mode in which the corpses of monsters do not disappear at all.

**Comment by SomeRandomNoobKekeke** [comment]
you can edit corpsedespawndelay in config_player.xml

and see how it works here https://github.com/FakeFishGames/Barotrauma/blob/master/Barotrauma/BarotraumaShared/SharedSource/Characters/Character.cs#L3639

**Reply by pavel4win** [reply]
Than you very much. This is really helps me. I'm happy now. No unexpected despawns anymore.

**Reply by BlackMoons** [reply]
Looks like you need to edit both   corpsedespawndelay and corpsespersubdespawnthreshold.

Mine was set to 600 and 5 respectively, but iv seen corpses despawn MUCH faster then 10 minutes. Looks like its something like any time you have more corpses, the decay timer reduces for *all* of them extremely fast. Looks like AI's also decay 2x as fast.

For example, if you kill 15 AI monsters within 1 minute, the following will result:

Each monster will tick down to despawn at 10x normal speed, resulting in all the monsters losing 9 minutes of 'decay time' in roughly 30 seconds. 

Over the next minute, every monster will then despawn in rapid succession leaving you nothing to loot. I feel this is a very flawed system, as it means a large swarm all despawns within 1~2 minutes. 

Even if you kill say, 10 creatures, the result is all 10 creatures decay at 10x speed for 1 minute, then start rapidly despawning over the next minute. 

Judging from the code, NPC's will also not decay if less then corpsespersubdespawnthreshold corpses exist. 

Recommendation to users: Increase corpsespersubdespawnthreshold from the *5* its currently at if you want more corpses around (Although that will result in more never decaying!), or consider that while corpsedespawndelay is in seconds, it may easily tick down up to (and even over) 20x faster for monsters once you have a few corpses around, so values like 1000~4000 might be needed to keep corpses around for a few minutes.

Recommendation to developer: Some randomization and/or max level for the increase in decay time, such that the majority of enemies never disappear faster then a few minutes (Or perhaps a 'min' decay time?)

Maybe a 'max' decay time would also be a good idea, so that all enemy AI corpses always decay eventually? (Right now, if less then corpsespersubdespawnthreshold, they will never despawn)  Iv gotten enemies stuck on my sub till round end, unable to move them, blocking controls/etc. (Corpses that spawn dead/friendly, AKA for missions/wrecks/etc could ignore this timer).

**Reply by SomeRandomNoobKekeke** [reply]
Just set it to 100000 and use [despawn button](https://steamcommunity.com/sharedfiles/filedetails/?id=3222992798)
![barodev small](https://github.com/FakeFishGames/Barotrauma/assets/122838333/e9507ccb-8a55-485d-afa8-88fc9754a87a)
